api/src/repositories/survey-repository.ts

```diff
@@ -815,4 +897,100 @@ export class SurveyRepository extends BaseRepository {
 
     await this.connection.sql(sqlStatement);
   }
+
+  /**
+   * Inserts a survey occurrence submission row.
+   *
+   * @param {IObservationSubmissionInsertDetails} submission The details of the submission
+   * @return {*} {Promise<{ submissionId: number }>} Promise resolving the ID of the submission upon successful insertion
+   */
+  async insertSurveyOccurrenceSubmission(
+    submission: IObservationSubmissionInsertDetails
+  ): Promise<{ submissionId: number }> {
+    defaultLog.debug({ label: 'insertSurveyOccurrenceSubmission', submission });
+    const queryBuilder = getKnex()
+      .table('occurrence_submission')
+      .insert({
+        ['input_file_name']: submission.inputFileName,
```

Comment:
Are the square brackets needed?

Reply:
Just a stylistic choice; Agreebly, not needed.

Reply:
Not a big deal either way then. I would maybe lean towards removing the brackets and the quotes then, just to simplify it a bit (especially in the context of knex, where things are already a bit more complicated)